From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Mon, 15 Jul 2024 14:13:39 -0500
Subject: [PATCH] link libmariadb

libmariadb exports many of the symbols that are present in mariadbclient.a (an internal static lib). Let's add this to the link targets for client binaries in order to reduce package size a bit

---
 client/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/client/CMakeLists.txt b/client/CMakeLists.txt
index a8e80bc8b5..bc747899ef 100644
--- a/client/CMakeLists.txt
+++ b/client/CMakeLists.txt
@@ -33,7 +33,7 @@ INCLUDE_DIRECTORIES(BEFORE
 ## We will need libeay32.dll and ssleay32.dll when running client executables.
 COPY_OPENSSL_DLLS(copy_openssl_client)
 
-SET(CLIENT_LIB mariadbclient mysys)
+SET(CLIENT_LIB libmariadb mariadbclient mysys)
 
 ADD_DEFINITIONS(${SSL_DEFINES})
 MYSQL_ADD_EXECUTABLE(mariadb completion_hash.cc mysql.cc readline.cc
